install/Knative-custom-install.md

```diff
@@ -0,0 +1,222 @@
+# Performing a Custom Knative Installation
+
+Use this guide to perform custom installations of Knative on your existing
+Kubernetes clusters. Knative's pluggable components allow you to
+install only what you want in each of your clusters.
+
+The steps covered in the guide are for advanced operators who want to
+customize each Knative installation. Installing individual Knative components
+requires that you to run multiple and separate installation commands.
+
+## Before you begin
+
+* If you are new to Knative, you should instead [install one of the default
+  installation packages](./README.md). Follow the host specific Knative installation
+  instructions for the default package to help you get up and running with the
+  least effort. Installing a default package also ensures that you can run all the
+  [Knative Serving Samples](https://github.com/knative/docs/blob/master/serving/samples/README.md).
+
+* The following steps use `bash` for a MacOS or Linux environment; for Windows,
+  some commands might need adjustment.
+
+* This installation guide assumes that you have an existing Kubernetes cluster,
+  on which you're comfortable installing and running _alpha_ level software.
+
+* Kubernetes requirements:
+
+  * Knative requires that the version of your Kubernetes cluster is v1.10 or newer.
+
+  * Your Kubernetes cluster must also have the
+    [MutatingAdmissionWebhook admission controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#how-do-i-turn-on-an-admission-controller)
+    enabled.
+
+  * The version of your
+    [`kubectl` CLI tool](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+    must be v1.10 or newer.
+
+## Installing Istio
+
+Knative depends on [Istio](https://istio.io/docs/concepts/what-is-istio/) for
+the service mesh. You can choose the default package to install the Istio service mesh with
+[automatic sidecar injection](https://istio.io/docs/setup/kubernetes/sidecar-injection).
+You should install the default package if you are just getting started with Knative.
+
+Alternatively, you can choose to install Istio with manual sidecar injection or with support for
+[Custom Resource Definitions (CRD)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/).
+
+### Choosing an Istio installation package
+
+You can choose from the following installation packages to install the Istio
+service mesh on you cluster:
+
+| Istio Package Filename | Description       |
+| ---------------------- | ----------------- |
+| [`istio.yaml`][a]             | Default: Configured for Automatic Sidecar Injection. |
+| [`istio-crds.yaml`][b]        | Support for Custom Resource Definitions. **Q: This have to do with "Pilot"? Why choose this?** |
```

Comment:
As this document is mainly for customize install for Knative, so I suggest we use select a default Istio install here to simplify the installation, how about only mention `istio.yaml` with sidecar autoinjection enabled?

Reply:
It is currently labelled as "default" but per a recent discussion with @mattmoor, the preferred and more reliable install path is to initially install `istio-crds.yaml` to ensure that required k8s resources (services?) are up and running. Apparently some are seeing issues if the customresourcedefinitions try to get created before k8s is ready.